src/data_generators/synthetic_data_generator.py

```diff
@@ -0,0 +1,95 @@
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Generates a DataDesign from SyntheticDataDesignConfig."""
+
+from absl import app
+from absl import flags
+import math
+import numpy as np
+from typing import List
+from wfa_planning_evaluation_framework.data_generators.data_design import DataDesign
+from wfa_planning_evaluation_framework.data_generators.data_set import DataSet
+from wfa_planning_evaluation_framework.data_generators.data_set_parameters import (
+    DataSetParameters, GeneratorParameters)
+from wfa_planning_evaluation_framework.data_generators.pricing_generator import PricingGenerator
+from wfa_planning_evaluation_framework.data_generators.fixed_price_generator import FixedPriceGenerator
+from wfa_planning_evaluation_framework.data_generators.impression_generator import ImpressionGenerator
+from wfa_planning_evaluation_framework.data_generators.homogeneous_impression_generator import HomogeneousImpressionGenerator
+from wfa_planning_evaluation_framework.data_generators.publisher_data import PublisherData
+from wfa_planning_evaluation_framework.data_generators.test_synthetic_data_design_config import TestSyntheticDataDesignConfig
+from wfa_planning_evaluation_framework.data_generators.synthetic_data_design_config import SyntheticDataDesignConfig
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('output_folder', 'TestDataDesign', 'Output Folder.')
+flags.DEFINE_string('data_design_config', 'TestConfig', 'Data Desgin Config.')
+flags.DEFINE_integer('random_seed', 1, 'Seed for the np.random.Generator.')
+
+name_to_config_dict = {'test': TestSyntheticDataDesignConfig}
+
+
+class SyntheticDataGenerator():
+  """Generates a DataDesign with synthetic data derived from parameters.
+
+    This class translates a SyntheticDataDesignConfig object to a DataDesign by
+    constructing the underlying objects and managing the publisher sizes.
+    """
+
+  def __init__(self, output_folder: str, random_seed: int,
+               config: SyntheticDataDesignConfig):
+    self._config = config
+    self._random_generator = np.random.default_rng(random_seed)
+    self._output_folder = output_folder
+
+  def __call__(self) -> DataDesign:
+    data_design = DataDesign(dirpath=self._output_folder)
+    for data_set_parameters in self._config.get_data_set_params_list(
+        self._random_generator):
+      data_design.add(self.generate_data_set(data_set_parameters))
+    return data_design
+
+  def generate_data_set(self, params: DataSetParameters) -> DataSet:
+    publishers = []
+    publisher_size = params.largest_publisher_size
+    publisher_size_decay_rate = params.largest_to_smallest_publisher_ratio**(
```

Comment:
Can you add a unit test for the case when the number of publishers is 1?

Reply:
:) thank you for spotting this!